tests/sources/test_ads.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+# (C) Copyright 2020 ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+#
+
+import os
+
+import pytest
+
+from earthkit.data import from_source
+
+NO_ADS = not os.path.exists(os.path.expanduser("~/.adsapirc"))
```

Comment:
`NO_ADS` should be defined in `testing.py`, next to `NO_CDS`, so that other test fixtures can use it.